### PR TITLE
Add additional test for `unnecessary_lambda_linter()`

### PR DESCRIPTION
**Pull request by IndrajeetPatil**

### CODE REVIEW

**Comment by MichaelChirico** [comment]
oh, I'm surprised this passes. can you check TODOs in the linter code, I seem to recall there's a TODO about exactly this test case...

**Comment by IndrajeetPatil** [comment]
This one?

https://github.com/r-lib/lintr/blob/75882b5a46ebacdbe0a2ec898f726fe29e7bc1f5/R/unnecessary_lambda_linter.R#L31-L34

**Comment by MichaelChirico** [comment]
yes, shouldn't this condition be failing your new test 😵‍💫

https://github.com/r-lib/lintr/blob/75882b5a46ebacdbe0a2ec898f726fe29e7bc1f5/R/unnecessary_lambda_linter.R#L45

**Comment by IndrajeetPatil** [comment]
The test should fail, right? 

Here `count(SYMBOL_FORMALS)` is `2`, and so `count(SYMBOL_FORMALS) = 1` condition is not met and xpath doesn't catch anything. So no lint.

File: tests/testthat/test-unnecessary_lambda_linter.R [hunk 1]
```diff
@@ -12,6 +12,9 @@ test_that("unnecessary_lambda_linter skips allowed usages", {
   # at any nesting level
   expect_lint("parLapply(cl, x, function(xi) foo(xi, 2, bar(baz(xi))))", NULL, linter)
 
+  # multi-argument case
+  expect_lint("lapply(l, function(x, y) setdiff(x, y))", NULL, linter)
```
Comment (MichaelChirico) [inline tests/testthat/test-unnecessary_lambda_linter.R:16]: oh I see now. I wouldn't add this test. the code as written can't run (y has no default), and any working version can be replaced without lambda.

no sense adding an incorrect test

Comment (IndrajeetPatil) [reply tests/testthat/test-unnecessary_lambda_linter.R:16]: Can replace it with this one:

``` r
l <- list("x" = 1:3, "y" = c(1:3, NA))

lapply(l, function(x, na.rm = TRUE) mean(x, na.rm = na.rm))
#> $x
#> [1] 2
#> 
#> $y
#> [1] 2
```

<sup>Created on 2022-10-03 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

File: tests/testthat/test-unnecessary_lambda_linter.R [hunk 2]
```diff
@@ -13,7 +13,7 @@ test_that("unnecessary_lambda_linter skips allowed usages", {
   expect_lint("parLapply(cl, x, function(xi) foo(xi, 2, bar(baz(xi))))", NULL, linter)
 
   # multi-argument case
-  expect_lint("lapply(l, function(x, y) setdiff(x, y))", NULL, linter)
+  expect_lint("lapply(l, function(x, na.rm = TRUE) mean(x, na.rm = na.rm))", NULL, linter)
```
Comment (MichaelChirico) [inline tests/testthat/test-unnecessary_lambda_linter.R:16]: I still don't see why you'd need a lambda here.

lapply(l, mean, na.rm=TRUE) does the same.

Comment (IndrajeetPatil) [reply tests/testthat/test-unnecessary_lambda_linter.R:16]: Fair enough.

**Pull request closed without merging**